File: 238 - Product of Array Except Self/array.go
func productExceptSelf(nums []int) []int {

	response := []int{}
	left := make([]int, len(nums))
	right := make([]int, len(nums))
	result := 1

	for i := 0; i < len(nums); i++ {
		result *= nums[i]
		left[i] = result
	}

	result = 1

	for j := len(nums) - 1; j >= 0; j-- {
		result *= nums[j]
		right[j] = result
	}

	for j := 0; j < len(nums); j++ {
		res := 0
		if j == 0 {
			res = right[j+1]
		} else if j == len(nums)-1 {
			res = left[j-1]
		} else {
			res = left[j-1] * right[j+1]
		}
		response = append(response, res)
	}

	return response
}

File: 238 - Product of Array Except Self/array.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        response: list[int] = []
        result = 1
        left: list[int] = []
        right: list[int] = []

        for num in nums:
            result *= num
            left.append(result)
        
        result = 1

        for i in range(len(nums) -1, -1, -1):
            result *= nums[i]
            right.insert(0,result)

        for i in range(len(nums)):
            if i == 0:
                response.append(right[i+1])
            elif i == len(nums) -1:
                response.append(left[i-1])
            else:
                response.append(left[i-1] * right[i+1])

        return response
        